### PR TITLE
CMR-6875 codecov configuration

### PR DESCRIPTION
**Pull request by jaybarra**
CMR-6875

added codecov.yml file to fix some path changes in codecov

How to test said change
look at the PR branch in codecov and see that the files can be browsed in the correct location from codecov including syntax highlighting for coverage on the code

[codecov commit file browser](https://codecov.io/gh/nasa/eo-metadata-tools/tree/f17c6001543d1fa2b45b22c171e1916a805076b0)

Verify tests continue to run locally
```
cd CMR/python
./runme.sh -t
```

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -23,8 +23,9 @@ jobs:
       - name: Checkout code
         uses: actions/checkout@v2
       - name: Run Unit Tests
-        run: ./runme.sh -t
-        working-directory: ./CMR/python
+        run: |
+          pip3 install coverage
+          coverage run --source=CMR/python -m unittest discover -s CMR/python
```
Comment (jaybarra) [inline .github/workflows/tests.yml:28]: `coverage` likes to generate relative paths which confused codecov. The native codecov path fixes were ignored. Run from the root, the paths for the python code are now correct in the report, letting codecov map the coverage to the source so the reporting is now browseable inside codecov.

File: CMR/python/test/cmr/search/test_common.py [hunk 2]
```diff
@@ -185,12 +185,14 @@ def test_cmr_query_url(self):
     @patch('urllib.request.urlopen')
     def test_scroll(self, urlopen_mock):
         """ Test the scroll clear function to see if it returns an error or not"""
-        recorded_data_file = 'test/data/cmr/common/scroll_good.json'
+        recorded_data_file = os.path.join (os.path.dirname (__file__),
```
Comment (jaybarra) [inline CMR/python/test/cmr/search/test_common.py:188]: Using `os.path.dirname (__file__)` resolves the resource location problem when running coverage from anywhere other than the `CMR/python` directory.

File: .github/workflows/tests.yml (hunk 1 above)
Comment (mschmele) [reply .github/workflows/tests.yml:28]: Maybe we want to leave a comment detailing that in this file

**Review by mschmele — APPROVED** [review]

File: .github/workflows/tests.yml (hunk 1 above)
Comment (jceaser) [reply .github/workflows/tests.yml:28]: is this test something that you would not run locally? the project documents using runme.sh for common tasks, but if this is something that only gets run on github, then I can get behind it.

Comment (jceaser) [reply .github/workflows/tests.yml:28]: got my answer, this call results in different output then what a user would get from runme.sh, this one is machine readable.

**Review by jceaser — APPROVED** [review]

**Pull request merged**